src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -44,7 +44,11 @@
    Connection connection;
    long lastAccessed;
    long lastBorrowed;
+
+   //should be used with -XX:-RestrictContended JVM arg
+   @sun.misc.Contended
```

Comment:
Maybe not, but unsafe, for example, used by many other libs and it's OK.
I've tested many times with contented annotation and without it. The difference is about ~2-5%.
But for case when threads = cpu cores this will be mandatory condition - reduce false sharing.
Only thing than save from it is random location of objects in memory because of they are added asynchronously and with random delays. So JVM locate them randomly in the heap often. But sometimes they can be located sequentally in memory. 
If not sun.misc.Contended than int var1,var2,vr3,var4 .... etc. 

The killer feature is recheck object state after set it free.